pallets/registry/src/benchmarking.rs

```diff
@@ -62,15 +62,189 @@ benchmarks! {
+
+    let raw_registry: Vec<u8> = (0u8..u8::MAX).cycle().take(T::MaxEncodedRegistryLength::get().try_into().unwrap()).collect();
+    let registry: InputRegistryOf::<T> = BoundedVec::try_from(raw_registry)
+        .expect("Test Registry should fit into the expected input length of the test runtime");
+    let digest = <T as frame_system::Config>::Hashing::hash(&registry[..]);
+    let id_digest = <T as frame_system::Config>::Hashing::hash(
+        &[&registry.encode()[..], &did.encode()[..]].concat()[..],
+    );
+    let registry_id = Ss58Identifier::to_registry_id(&(id_digest).encode()[..]).unwrap();
+
+    Pallet::<T>::create(origin.clone(), registry, None).expect("Should create a registry entry");
+    Pallet::<T>::archive(origin.clone(), registry_id.clone()).expect("Should archive the registry");
+
+}: _<T::RuntimeOrigin>(origin, registry_id.clone())
+
+verify {
+    assert_last_event::<T>(Event::Restore { registry: registry_id, authority: did }.into());
+}
+
+add_admin_delegate {
+    let l in 1 .. MAX_REGISTRY_SIZE;
+
+    let caller: T::AccountId = account("caller", 0, SEED);
+    let did: T::RegistryCreatorId = account("did", 0, SEED);
+    let delegate: T::RegistryCreatorId = account("delegate", 0, SEED);
+
+    let origin =  <T as Config>::EnsureOrigin::generate_origin(caller.clone(), did.clone());
+
+    let raw_registry: Vec<u8> = (0u8..u8::MAX).cycle().take(T::MaxEncodedRegistryLength::get().try_into().unwrap()).collect();
+    let registry: InputRegistryOf::<T> = BoundedVec::try_from(raw_registry)
+        .expect("Test Registry should fit into the expected input length of the test runtime");
+    let id_digest = <T as frame_system::Config>::Hashing::hash(
+        &[&registry.encode()[..], &did.encode()[..]].concat()[..],
+    );
+    let registry_id = Ss58Identifier::to_registry_id(&(id_digest).encode()[..]).unwrap();
+
+    Pallet::<T>::create(origin.clone(), registry, None).expect("Should create a registry entry");
+
+    let origin_creator =  <T as Config>::EnsureOrigin::generate_origin(caller.clone(), did.clone());
+   
+}: _<T::RuntimeOrigin>(origin, registry_id.clone(), delegate.clone())
+verify {
+    // assert_eq!(authorization.registry_id, registry_id, "Registry ID should match");
+    // assert_eq!(authorization.delegate, delegate, "Delegate should match");
+     let authorization_id = Ss58Identifier::to_authorization_id(
+        &(<T as frame_system::Config>::Hashing::hash(
+            &[&registry_id.encode()[..], &delegate.encode()[..], &did.encode()[..]].concat()[..],
+        )).encode()[..]
+    ).expect("Authorization ID should be generated");
+
+    let authorization = <Authorizations<T>>::get(&authorization_id).expect("Authorization should exist");
+
+    assert_last_event::<T>(Event::AddAuthorization {
+        registry: registry_id,
+        authorization: authorization_id,
+        delegate: delegate,
+    }.into());
```

Comment:
assert_last_event would be a good enough check, as it checks for the call success. We can remove the storage checks

Reply:
okay

Reply:
Done